### ISSUE TITLE
Empty index_list 

### ISSUE THREAD

**Issue by gold-Soul** [hslyu/GIF#2]
I choice the selection scheme “HighestKGradients” and run the code as follows: 
`net_parser.register_hooks()`
`target_loss = (criterion(outputs, labels))`
`target_loss.backward(retain_graph=True)`
`index_list = net_parser.get_parameters()`

But I found that the index_list is none and the function "hook" in function "generated_hook" from the file "HighestKGradients.py" didn't run.

![Figure1](https://github.com/user-attachments/assets/35db1f79-023e-454f-94a6-382f1a0b176f)
 
![Figure2](https://github.com/user-attachments/assets/44e68371-f371-4b11-9660-3f90afb03cb5)

**Comment by hslyu** [comment]
I wonder how you configure the parameter ratio and the input tensors.
It looks like you have customized the code. I wonder the original code returns the same empty array either.

**Comment by gold-Soul** [comment]
-1. I have test it that the register hook function 'hook_fn = self.generate_hook(start_index)'  has been run. 
-2. The parameter ratio was set as 0.2.
-3. The input tensors include the loss of a subset of a target class and the loss of a data point.
-4. The hook function didn't run during the feed-forward and back-propagation.
-5. The customized code about GIF compution and hook is highly similar to the original code in GIF-main/scripts/table1-parameter_selection_comparison_cifar10_0.ipynb. But I didn't run this ipynb file

**Comment by gold-Soul** [comment]
> I wonder how you configure the parameter ratio and the input tensors. It returns the empty array if hooks are not registered or the parameter ratio is zero.
> 
> Can you test whether the hook functions are called during the feed-forward and back-propagation? It looks like you have customized the code. I wonder the original code returns the same empty array either.

**Comment by hslyu** [comment]
The hook functions are designed as callback functions that collect parameter indices when layerwise inputs/outputs (or gradient inputs/outputs) goes through their own layer. 
Thus, if the hook functions are not running during the feed-forward and back-propagation, the ``get_parameters`` method should return the empty list.
